File: python/Numbers/Fibonacci/fibonacci.py
def calc_fib(n, cb = lambda x: x, every = True):
    if n <= 0:
        return
    
    a = 0
    b = 1
    c = 0

    for i in range(2, n + 1):
        if every:
            cb(c)
        c = a + b
        a, b = b, c

    cb(c)

File: javascript/Numbers/Fibonacci/fibonacci.js
const BigNumber = require('bignumber.js')

exports.calculateFibonacciNumber = function (n, cb, every = true) {
  if (n.lte(0)) {
    return
  }

  let a = new BigNumber(0)
  let b = new BigNumber(1)
  let c = new BigNumber(0)

  for (let i = 2; i <= n; i++) {
    if (every) {
      cb?.apply(this, [c.toString()])
    }
    c = a.plus(b)
    a = b
    b = c
  }

  cb?.apply(this, [c.toString()])
}
